presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/PredicatePushDown.java

```diff
@@ -458,15 +460,24 @@ private OuterJoinPushDownResult processLimitedOuterJoin(Expression inheritedPred
                 }
             }
 
-            // See if we can push down any outer or join predicates to the inner side
```

Comment:
@martint please take a look. Here. I am not 100% sure I did not mess something up here.
My intention was to rewrite joinPredicate in context of inner symbols.
Did I break some optimization by removing `joinPredicate` from `EqualityInference.nonInferrableConjuncts(and(outerEffectivePredicate, joinPredicate)` ?